Named tensor support for: atan2, output_nr, detach{_}, requires_grad_

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#26543 Named tensor support for: atan2, output_nr, detach{_}, requires_grad_**

Also adds a test for logical_xor (it already had named tensor support
but there was no test)

Test Plan:
- [namedtensor ci]

Differential Revision: [D17501403](https://our.internmc.facebook.com/intern/diff/D17501403)

zou3519 committed:
Named tensor support for: atan2, output_nr, detach{_}, requires_grad_

Also adds a test for logical_xor (it already had named tensor support
but there was no test)

Test Plan:
- [namedtensor ci]

zou3519 committed:
Update on "Named tensor support for: atan2, output_nr, detach{_}, requires_grad_"


Also adds a test for logical_xor (it already had named tensor support
but there was no test)

Test Plan:
- [namedtensor ci]

Differential Revision: [D17501403](https://our.internmc.facebook.com/intern/diff/D17501403)

zou3519 committed:
Update on "Named tensor support for: atan2, output_nr, detach{_}, requires_grad_"


Also adds a test for logical_xor (it already had named tensor support
but there was no test)

Test Plan:
- [namedtensor ci]

Differential Revision: [D17501403](https://our.internmc.facebook.com/intern/diff/D17501403)

zou3519 committed:
Update on "Named tensor support for: atan2, output_nr, detach{_}, requires_grad_"


Also adds a test for logical_xor (it already had named tensor support
but there was no test)

Test Plan:
- [namedtensor ci]

Differential Revision: [D17501403](https://our.internmc.facebook.com/intern/diff/D17501403)

nairbv commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -4899,9 +4903,11 @@
     CUDA: legacy::cuda::_th_dist
 
 - func: atan2.out(Tensor self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)
+  supports_named_tensor: True
```
do we need any new named tensor tests for these?

zou3519 committed:
Update on "Named tensor support for: atan2, output_nr, detach{_}, requires_grad_"


Also adds a test for logical_xor (it already had named tensor support
but there was no test)

Test Plan:
- [namedtensor ci]

Differential Revision: [D17501403](https://our.internmc.facebook.com/intern/diff/D17501403)
